internal/runners/exec/exec.go

```diff
@@ -122,25 +121,22 @@ func (s *Exec) Run(params *Params, args ...string) (rerr error) {
 		rtTarget = target.NewProjectTarget(proj, nil, trigger)
 	}
 
-	err = buildscript.Sync(proj, nil, s.out, s.auth)
-	if err != nil {
-		return locale.WrapError(err, "err_update_build_script")
-	}
-
 	rtusage.PrintRuntimeUsage(s.svcModel, s.out, rtTarget.Owner())
 
 	s.out.Notice(locale.Tl("operating_message", "", projectNamespace, projectDir))
 
-	rt, err := runtime.New(rtTarget, s.analytics, s.svcModel)
-	if err != nil {
+	rt, err := runtime.New(rtTarget, s.analytics, s.svcModel, s.auth)
+	if err != nil && !runtime.IsNeedsStageError(err) {
 		if !runtime.IsNeedsUpdateError(err) {
 			return locale.WrapError(err, "err_activate_runtime", "Could not initialize a runtime for this project.")
 		}
 		pg := runbits.NewRuntimeProgressIndicator(s.out)
 		defer rtutils.Closer(pg.Close, &rerr)
-		if err := rt.Update(s.auth, pg); err != nil {
+		if err := rt.Update(pg); err != nil {
 			return locale.WrapError(err, "err_update_runtime", "Could not update runtime installation.")
 		}
+	} else if runtime.IsNeedsStageError(err) {
+		s.out.Notice(locale.T("notice_stage"))
```

Comment:
This feels like an awkward mix of patterns. Above we are checking `err != nil` and then handling cases of err not being nil. This else statement should IMO be part of those cases.